### ISSUE TITLE
test_point_range_captures not working

### ISSUE THREAD

**Issue by lunixbochs** [tree-sitter/py-tree-sitter#105]
With start_point/end_point in `test_byte_range_captures`, we're getting a capture result outside the point range `(1,0) - (2,0)`:
```
[(<Node type=identifier, start_point=(0, 4), end_point=(0, 7)>, 'func-def'), (<Node type=identifier, start_point=(1, 2), end_point=(1, 5)>, 'func-call')]
```

I don't see the same issue with the byte range query from `test_byte_range_captures`:
```
[(<Node type=identifier, start_point=(1, 2), end_point=(1, 5)>, 'func-call'), (<Node type=identifier, start_point=(2, 4), end_point=(2, 7)>, 'func-def')]
```

@maxbrunsfeld can you take a look at this? I confirmed that tree sitter is querying with the correct ranges, so I think this relies on some upstream tree sitter behavior I don't understand.

https://github.com/tree-sitter/py-tree-sitter/blob/3bc64bb2603559a1fe45a4526eb797ae66fe90b7/tests/test_tree_sitter.py#L863-L880

**Comment by maxbrunsfeld** [comment]
Yeah, this is a Tree-sitter bug; there's a small bit of logic that was only implemented for `byte_range`, and `point_range`. 

Right now, the `point_range` is used to filter out entire *matches* (at least one node in the pattern must intersect the point range), but it doesn't filter out individual *captures* within a match. That additional per-capture filtering is currently performed only based on the byte range. It's done [here](https://github.com/tree-sitter/tree-sitter/blob/465ceead0faf3a0aab311806795664a2c2fad5de/lib/src/query.c#L3790-L3795). It should be pretty easy to also take into account the `point_range`.

**Comment by MrPrezident** [comment]
I'm running into this issue. Looks like this is not fixed yet?

**Comment by MrPrezident** [comment]
Looks like this requires a pretty simple fix so I created a pull request. I tested it out on my project, and that seems to fix the issue.

**Comment by maxbrunsfeld** [comment]
Thanks @MrPrezident; I left a comment on your PR.

**Comment by MrPrezident** [comment]
Thanks @maxbrunsfeld I replied to your comment and updated the PR branch.
